backend/question-service/src/controllers/question.controller.ts

```diff
@@ -0,0 +1,76 @@
+import {
+    createQuestion,
+    findOneQuestionById,
+    findOneQuestionByTitle,
+    findPaginatedQuestions,
+    findQuestionCount,
+} from '../models/question.repository'
+
+import { CreateQuestionDto } from '../types/CreateQuestionDto'
+import { QuestionDto } from '../types/QuestionDto'
+import { Response } from 'express'
+import { TypedRequest } from '../types/TypedRequest'
+import { ValidationError } from 'class-validator'
+
+export async function handleCreateQuestion(
+    request: TypedRequest<CreateQuestionDto>,
+    response: Response
+): Promise<void> {
+    const createDto = CreateQuestionDto.fromRequest(request)
+    const errors = await createDto.validate()
+    if (errors.length) {
+        const errorMessages = errors.map((error: ValidationError) => `INVALID_${error.property.toUpperCase()}`)
+        response.status(400).json(errorMessages).send()
+        return
+    }
+
+    const duplicate = await findOneQuestionByTitle(createDto.title)
+    if (duplicate) {
+        response.status(409).json('DUPLICATE_TITLE').send()
+        return
+    }
+
+    const question = await createQuestion(createDto)
+    const dto = QuestionDto.fromModel(question)
+    response.status(201).json(dto).send()
+}
+
+export async function handleGetPaginatedQuestions(request: TypedRequest<void>, response: Response): Promise<void> {
+    const page = parseInt(request.query.page as string)
```

Comment:
you can create a new generic interface similar to TypedRequest to specifically handle pagination requests

Reply:
I'll add the changes on my end in the new branch